pymeasure/instruments/thyracont/smartline_v1.py

```diff
@@ -0,0 +1,144 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments.validators import strict_discrete_set
+from pymeasure.instruments import Instrument
+
+
+def calculateChecksum(msg):
+    """Calculate a 1 bytes checksum mapped to a printable ASCII character.
+
+    The checksum is calculated by the sum of the decimal values of each message
+    character modulo 64 + 64.
+
+    :param string msg: message content
+    :returns: calculated checksum
+    :rtype: string
+    """
+    return chr(sum(map(ord, msg)) % 64 + 64)
```

Comment:
We could move it into some other file (init? @bilderbuchi ) for use of both instruments.
On the other hand, the method is not too large, such that it can remain in both instruments.

I think a couple of tests would be good.

Reply:
I added some tests to check the `calculate_checksum` functionality. Especially since we now have one function which is tested I think even though its a short method it makes sense to reuse it in both v1 and v2 files.

Reply:
Let's keep it in v1, which is a small file (with only a few dependencies) anyways.